src/map.cpp

```diff
@@ -3005,71 +2987,89 @@ MapBlock* ServerMap::loadBlock(v3s16 blockpos)
 {
 	DSTACK(FUNCTION_NAME);
 
+	bool created_new = (getBlockNoCreateNoEx(blockpos) == NULL);
+
 	v2s16 p2d(blockpos.X, blockpos.Z);
 
 	std::string ret;
 	dbase->loadBlock(blockpos, &ret);
 	if (ret != "") {
 		loadBlock(&ret, blockpos, createSector(p2d), false);
-		return getBlockNoCreateNoEx(blockpos);
-	}
-	// Not found in database, try the files
-
-	// The directory layout we're going to load from.
-	//  1 - original sectors/xxxxzzzz/
-	//  2 - new sectors2/xxx/zzz/
-	//  If we load from anything but the latest structure, we will
-	//  immediately save to the new one, and remove the old.
-	int loadlayout = 1;
-	std::string sectordir1 = getSectorDir(p2d, 1);
-	std::string sectordir;
-	if(fs::PathExists(sectordir1))
-	{
-		sectordir = sectordir1;
-	}
-	else
-	{
-		loadlayout = 2;
-		sectordir = getSectorDir(p2d, 2);
-	}
-
-	/*
-		Make sure sector is loaded
-	*/
-
-	MapSector *sector = getSectorNoGenerateNoEx(p2d);
-	if(sector == NULL)
-	{
-		try{
-			sector = loadSectorMeta(sectordir, loadlayout != 2);
-		}
-		catch(InvalidFilenameException &e)
+	} else {
+		// Not found in database, try the files
+
+		// The directory layout we're going to load from.
+		//  1 - original sectors/xxxxzzzz/
+		//  2 - new sectors2/xxx/zzz/
+		//  If we load from anything but the latest structure, we will
+		//  immediately save to the new one, and remove the old.
+		int loadlayout = 1;
+		std::string sectordir1 = getSectorDir(p2d, 1);
+		std::string sectordir;
+		if(fs::PathExists(sectordir1))
 		{
-			return NULL;
+			sectordir = sectordir1;
 		}
-		catch(FileNotGoodException &e)
+		else
 		{
-			return NULL;
+			loadlayout = 2;
+			sectordir = getSectorDir(p2d, 2);
 		}
-		catch(std::exception &e)
+
+		/*
+		Make sure sector is loaded
+		 */
+
+		MapSector *sector = getSectorNoGenerateNoEx(p2d);
+		if(sector == NULL)
 		{
-			return NULL;
+			try{
+				sector = loadSectorMeta(sectordir, loadlayout != 2);
+			}
+			catch(InvalidFilenameException &e)
+			{
+				return NULL;
```

Comment:
If you're catching these shouldn't they be logged? (otherwise why not just catch them all and return NULL?)